### ISSUE TITLE
Extracting element list into AdBlock/uBlock-compatible list possible?

### ISSUE THREAD

**Issue by katrinleinweber** [bengrosser/twitter-demetricator#7]
Hello and thanks for creating this extension :-)

As users in an environment where browser extension installs are limited to a few approved ones, some may want to get the core benefit of blocking [those `div` elements](https://github.com/bengrosser/twitter-demetricator/blob/e142de9043399c259103199ecf8b0642cd610dce/twitter-demetricator.user.js#L886-L960) (I guess) via an alternative route.

Would it be possible to extract that list into a separate file in a [compatible syntax](https://help.eyeo.com/en/adblockplus/how-to-write-filters)? With some pointers to the JS functions needed to achieve this, I'd be happy to try implementing this.

**Comment by bengrosser** [comment]
Hi Katrin. 

So the short answer is that you can find a list of CSS selectors I target in [td.css](https://github.com/bengrosser/twitter-demetricator/blob/e142de9043399c259103199ecf8b0642cd610dce/firefox/td.css). Focusing on implementing this list using something you already have access to (like an adblock filter list) would allow you to catch some of the metrics I hide with Twitter Demetricator. 

The longer answer is that, because of how Twitter works—and because of my stubborn intention to hide _only_ the metrics—the rest are more complicated. For example, when you see something like "Username and 8 others liked this" I often can't just hide that line or that number via CSS, but need to examine the line, find the metric, and then rewrite the sentence without the number. Other times I have to clone the parent element, hide the original, and then show my copy (with metrics hidden) This is because I am preserving the constantly changing metrics state so that users can toggle hiding on/off (an essential feature for new users in particular so they can ease into a demetricated social media landscape). There are a few other techniques for the various places in the interface that are more tricky.

Basically, if you want a brute force overly zealous demetricator, you can go through and find all the selectors I'm using in some way (both in that CSS file but also in the .js file you linked to) and just hide them and see what you get. Going this route you'll likely end up hiding more elements than you intended (and more than just the metrics), but it will get you started.

Hope that helps.
ben

**Comment by katrinleinweber** [comment]
Thanks for explaining :-)
